File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        freq={}
        for ch in s:
            freq[ch]=freq.get(ch,0)+1
        for ch in t:
            if freq.get(ch,0)==0:
                return False
            else:
                freq[ch]=freq.get(ch,0)-1
        if all(value==0 for value in freq.values()):
            return True
        else:
            return False
        
        

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.java

class Solution {
    public int findDuplicate(int[] nums) {
        HashMap<Integer,Integer> freq= new HashMap<>();
        for(int num:nums){
            freq.put(num,freq.getOrDefault(num,0)+1);
        }
        for(int num:nums){
            if(freq.get(num)>1){
                return num;
            }
        }
        return -1;
    }
}
